src/comm/MAVLinkProtocol.cc

```diff
@@ -159,6 +159,52 @@ void MAVLinkProtocol::resetMetadataForLink(LinkInterface *link)
     link->setDecodedFirstMavlinkPacket(false);
 }
 
+/**
+ * This method parses all outcoming bytes and log a MAVLink packet.
+ * @param link The interface to read from
+ * @see LinkInterface
+ **/
+
+void MAVLinkProtocol::logSentBytes(LinkInterface* link, QByteArray b){
+
+    uint8_t mavlinkChannel = link->mavlinkChannel();
+    static mavlink_message_t _sent_message;
+
+    for (int position = 0; position < b.size(); position++) {
+
+        if(mavlink_parse_char(mavlinkChannel, static_cast<uint8_t>(b[position]), &_sent_message, &_status)){
```

Comment:
You can't use the same mavlink channel here. It will screw up the link statistics stored in the mavlink status struct associated with the channel. In order to do this it will require a separately reserved mavlink channel for the decode. But that isn't really possible since it will cut in half the number of vehicles you can connect to. There is a way to change mavlink to allocate status structs instead of the using the fixed array. But QGC does not use that and it's a bit of work to switch to that, but doable.

Reply:
Sorry I didn't realize this would be a problem initially.

Reply:
Actually I don't need that test on my function as it's called by a signal emitted exactly when the message is sent. The buffer only contains the current message and starts by this message. Let me change this !